File: sortingAlgo/quickSort/quickSort.c
#include <stdio.h>
void swap(int array[] ,int index1, int index2);
int partition(int array[], int low, int high);
void quickSort(int array[], int low, int high);

int main() {
    int array[9] = {2,9,3,8,6,5,4,7,10};
    printf("Original array: ");
    for (int i = 0; i < 9; i++) {
        printf("%d ", array[i]);
    }

    quickSort(array, 0,9);

    printf("\nSorted array: ");
    for (int i = 0; i < 9; i++) {
        printf("%d ", array[i]);
    }

    return 0;
}

void swap(int array[], int index1, int index2) {
    int temp = array[index2];
    array[index2] = array[index1];
    array[index1] = temp;
}




int partition(int array[], int low, int high) {
    int pivot = array[low];
    int i = 0;
    int j = high;

    while(i < j) {
        do {
            i++;
        } while (array[i] <= pivot);

        do {
            j--;
        } while (array[j] > pivot);
        if (i < j) {
            swap(array, i, j);
        }
    }
    swap(array, low, j);
    return j;
}


void quickSort(int array[], int low, int high) {
    if (low < high) {
        int j = partition(array, low, high);
        quickSort(array, low, j);
        quickSort(array, j+1, high);
    }
}

File: sortingAlgo/quickSort/quickSort.cs
class QuickSort
{
  static int Partition(int[] array, int low,
                                  int high)
  {
    // Select a pivot point.
    int pivot = array[high];

    int lowIndex = (low - 1);

    // Reorder the collection.
    for (int j = low; j < high; j++)
    {
      if (array[j] <= pivot)
      {
        lowIndex++;

        int temp = array[lowIndex];
        array[lowIndex] = array[j];
        array[j] = temp;
      }
    }

    int temp1 = array[lowIndex + 1];
    array[lowIndex + 1] = array[high];
    array[high] = temp1;

    return lowIndex + 1;
  }

  static void Sort(int[] array, int low, int high)
  {
    if (low < high)
    {
      int partitionIndex = Partition(array, low, high);

      // Recursively continue sorting the array
      Sort(array, low, partitionIndex - 1);
      Sort(array, partitionIndex + 1, high);
    }
  }

  public static void Main()
  {
    int[] array = { 72, 12, 6, 33, 81, 97, 37, 59, 52, 1, 20 };
    int length = array.Length;

    Console.WriteLine("QuickSort");
    CommonFunctions.PrintInitial(array);
    Sort(array, 0, length - 1);

    CommonFunctions.PrintFinal(array);

    Console.ReadKey();
  }
}
